java/src/main/java/ai/rapids/cudf/nvcomp/BatchedLZ4Decompressor.java

```diff
@@ -0,0 +1,228 @@
+  }
+
+  /**
+   * Get the amount of ouptut storage required to decopmress a batch of buffers
+   * @param metadata   metadata retrieved from the compressed buffers
+   * @param numOutputs number of buffers in the batch
+   * @return amount in bytes of temporary storage space required to decompress the buffer batch
+   */
+  public static long[] getOutputSizes(BatchedMetadata metadata, int numOutputs) {
+    return NvcompJni.batchedLZ4DecompressGetOutputSize(metadata.getMetadata(), numOutputs);
+  }
+
+  /**
+   * Asynchronously decompress a batch of buffers
+   * @param inputs     buffers to decompress
+   * @param tempBuffer temporary buffer
+   * @param metadata   metadata retrieved from the compressed buffers
+   * @param outputs    output buffers that will contain the compressed results
+   * @param stream     CUDA stream to use
+   */
+  public static void decompressAsync(BaseDeviceMemoryBuffer[] inputs,
+                                     BaseDeviceMemoryBuffer tempBuffer, BatchedMetadata metadata,
+                                     BaseDeviceMemoryBuffer[] outputs, Cuda.Stream stream) {
+    int numBuffers = inputs.length;
+    if (outputs.length != numBuffers) {
+      throw new IllegalArgumentException("buffer count mismatch, " + numBuffers + " inputs and " +
+          outputs.length + " outputs");
+    }
+
+    long[] inputAddrs = new long[numBuffers];
+    long[] inputSizes = new long[numBuffers];
+    for (int i = 0; i < numBuffers; ++i) {
+      BaseDeviceMemoryBuffer buffer = inputs[i];
+      inputAddrs[i] = buffer.getAddress();
+      inputSizes[i] = buffer.getLength();
+    }
+
+    long[] outputAddrs = new long[numBuffers];
+    long[] outputSizes = new long[numBuffers];
+    for (int i = 0; i < numBuffers; ++i) {
+      BaseDeviceMemoryBuffer buffer = outputs[i];
+      outputAddrs[i] = buffer.getAddress();
+      outputSizes[i] = buffer.getLength();
+    }
+
+    NvcompJni.batchedLZ4DecompressAsync(inputAddrs, inputSizes,
+        tempBuffer.getAddress(), tempBuffer.getLength(), metadata.getMetadata(),
+        outputAddrs, outputSizes, stream.getStream());
+  }
+
+  /**
+   * Asynchronously decompress a batch of buffers
+   * @param inputs buffers to decompress
+   * @param stream CUDA stream to use
+   * @return output buffers containing compressed data corresponding to the input buffers
+   */
+  public static DeviceMemoryBuffer[] decompressAsync(BaseDeviceMemoryBuffer[] inputs,
+                                                     Cuda.Stream stream) {
+    int numBuffers = inputs.length;
+    long[] inputAddrs = new long[numBuffers];
+    long[] inputSizes = new long[numBuffers];
+    for (int i = 0; i < numBuffers; ++i) {
+      BaseDeviceMemoryBuffer buffer = inputs[i];
+      inputAddrs[i] = buffer.getAddress();
+      inputSizes[i] = buffer.getLength();
+    }
+
+    long metadata = NvcompJni.batchedLZ4DecompressGetMetadata(inputAddrs, inputSizes,
```

Comment:
`NvcompJni` methods are all package-private.  Applications cannot call them directly.  `BatchedMetadata` exists as a wrapper for the metadata internals and easier cleanup (since it implements `AutoCloseable` and leak detection tracking).  I did not use `BatchedMetadata` in this code to save some unnecessary object creation and leak detection registration.